collector/pg_stat_user_indexes.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2023 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package collector
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func init() {
+	registerCollector(statUserIndexesSubsystem, defaultDisabled, NewPGStatUserIndexesCollector)
+}
+
+type PGStatUserIndexesCollector struct {
+	log log.Logger
+}
+
+const statUserIndexesSubsystem = "stat_user_indexes"
+
+func NewPGStatUserIndexesCollector(config collectorConfig) (Collector, error) {
+	return &PGStatUserIndexesCollector{log: config.logger}, nil
+}
+
+var (
+	statUserIndexesIdxScan = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, statUserIndexesSubsystem, "idx_scan"),
```

Comment:
Counters should be plural and end in total.

```suggestion
		prometheus.BuildFQName(namespace, statUserIndexesSubsystem, "idx_scans_total"),
```